packages/shylu/shylu_dd/frosch/src/SchwarzOperators/FROSch_GeometricOverlappingOperator_decl.hpp

```diff
@@ -0,0 +1,280 @@
+
+        using XMatrix                 = typename SchwarzOperator<SC,LO,GO,NO>::XMatrix;
+        using XMatrixPtr              = typename SchwarzOperator<SC,LO,GO,NO>::XMatrixPtr;
+        using ConstXMatrixPtr         = typename SchwarzOperator<SC,LO,GO,NO>::ConstXMatrixPtr;
+
+        using ConstXCrsGraphPtr       = typename SchwarzOperator<SC,LO,GO,NO>::ConstXCrsGraphPtr;
+
+        using ParameterListPtr        = typename SchwarzOperator<SC,LO,GO,NO>::ParameterListPtr;
+
+    public:
+
+        /**
+         * @brief Constructor for the GeometricOverlappingOperator class.
+         *
+         * This is the default constructor that takes the (global) system_matrix 
+         * to construct the GeometricOverlappingOperator.
+         *
+         * To fine-tune the resulting preconditioner, several options can be controlled 
+         * via the provided parameter list:
+         * - **Combine Values in Overlap**: The method to combine values in overlap. Options are: Restricted, Averaging, Full. Default Restricted.
+         * - **Solver type**: The type of solver to use. The option is: Amesos2. 
+         * - **Solver**: The specific solver to use. The options are all direct solvers from Amesos2. Default KLU2
+         *
+         * @param k The (global) system_matrix.
+         * @param dualGraph The dual graph of the grid.
+         * @param overlap The overlap for the Schwarz preconditioner.
+         * @param parameterList The list of parameters for the GeometricOverlappingOperator.
+         */
+        GeometricOverlappingOperator(ConstXMatrixPtr  k,
+                                     int              overlap,
+                                     GraphPtr         dualGraph,
+                                     ParameterListPtr parameterList);
+
+        /**
+         * Not implemented. 
+         */
+        virtual int 
+        initialize() override;
+
+        /**
+         * @brief Initializes the GeometricOverlappingOperator based on the given overlap and the overlappingMap.
+         *
+         * This function creates the overlapping map for the cells (via calling 
+         * OverlappingOperator::buildOverlappingMap()). The map is computed based on the given 
+         * overlap and the dual graph, that was provided in the constructor of this class. 
+         * Moreover, this function assigns the overlapping DoF map to the internal stores it in 
+         * OverlappingMap_. 
+         *
+         * After assigning the overlapping dof map to this->OverlappingMap_, the other functions, 
+         * that are normally called in initialize are called (i.e. 
+         * OverlappingOperator::initializeOverlappingOperator() and 
+         * OverlappingOperator::updateLocalOverlappingMatrices_Symbolic).
+         *
+         * @param overlappingMap The overlapping dof map.
+         * @return Error code. 0 if successful.
+         */
+        int 
+        initialize(XMapPtr overlappingMap);
```

Comment:
I guess this comes from pattern matching, but I am not a fan of communicating errors via return codes.